scripts/checks/extract_scorecard_artifact.py

```
@@ -0,0 +1,130 @@
+        raise ValueError(f"artifact source does not exist: {source}")
+    ensure_non_symlink_path(source, path_kind="artifact path")
+    candidates: list[Path] = []
+    for path in sorted(
+        candidate for candidate in source.iterdir() if candidate.suffix == ".zip"
+    ):
+        ensure_non_symlink_path(path, path_kind="artifact path")
+        candidates.append(path)
+    if len(candidates) != 1:
+        raise ValueError(
+            f"expected exactly one Scorecard artifact zip in {source}, found {len(candidates)}"
+        )
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)